script/get_github_workflow_run_job_id.rb

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require "net/http"
+require "json"
+
+workflow_run_id = ARGV[0]
+run_attempt = ARGV[1]
+job_name = ARGV[2]
+
+uri =
+  URI.parse(
+    "https://api.github.com/repos/discourse/discourse/actions/runs/#{workflow_run_id}/attempts/#{run_attempt}/jobs",
+  )
+
+request = Net::HTTP::Get.new(uri)
+request["Accept"] = "application/vnd.github+json"
+request["X-Github-Api-Version"] = "2022-11-28"
+
+response =
+  Net::HTTP.start(uri.host, uri.port, use_ssl: uri.is_a?(URI::HTTPS)) do |http|
```

Comment:
Do we need any sort of error handling here? What's the impact of an exception at this point?

Reply:
Bad things will happen and if it does, my plan is to just give up on the workflow run. Things would have been way more predictable if only Github provided an easy way to grab the current job id. I'm hoping hitting Github's API from within Github's runner will be successful most of the time.